### PR TITLE
Plyta tygodnia page

### PR DESCRIPTION
**Pull request by julianuziemblo**
- add PlytaTygodniaPage widget that pulls data from radioaktywne.pl api
- refactor fetching and refreshing code into reusable RefreshableFetchWidget
- add multiple clarifying doc comments

obraz

### CODE REVIEW

File: mobile/lib/resources/fetch_data.dart [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+///
+/// The [fromJson] has to be a function
+/// that converts the provided JSON data to
+/// an object of type [T].
+/// For example, it can be [T]'s fromJson()
+/// constructor.
+///
+/// Throws [TimeoutException] if the fetching
+/// function exceeds given [timeout].
+Future<Iterable<T>> fetchData<T>(
```
Comment (lewandowski-jan) [inline mobile/lib/resources/fetch_data.dart:18]: I'd create get post methods for GET and POST

File: mobile/lib/pages/plyta_tygodnia_page.dart [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+      child: CircularProgressIndicator(
+        color: context.colors.highlightGreen,
+        strokeWidth: 5,
+      ),
+    );
+  }
+}
+
+/// Information about a single Plyta tygodnia.
+class PlytaTygodniaInfo {
```
Comment (lewandowski-jan) [inline mobile/lib/pages/plyta_tygodnia_page.dart:197]: I'd put data class in separate file

File: mobile/lib/pages/plyta_tygodnia_page.dart [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+                                child: CircularProgressIndicator(
+                                  color: context.colors.highlightGreen,
+                                  strokeWidth: 5,
+                                ),
+                              ),
+                            ),
+                          ),
+                errorBuilder: (context, child, loadingProgress) => Center(
+                  child: Text(
+                    'Wystąpił błąd podczas pobierania obrazu',
```
Comment (lewandowski-jan) [inline mobile/lib/pages/plyta_tygodnia_page.dart:98]: ohhh nooo, why are you not using https://github.com/RadioAktywne/ra-app/blob/master/mobile/lib/l10n/app_pl.arb

Comment (lewandowski-jan) [reply mobile/lib/pages/plyta_tygodnia_page.dart:98]: we can connect and I'll show you what's up

File: mobile/lib/main.dart [hunk 4]
```diff
@@ -108,153 +110,159 @@ class MainApp extends HookWidget {
...
                 ],
               ),
             ),
-            bottomNavigationBar: const RaBottomNavigationBar(),
-          ),
+          ],
         ),
-      ),
+
+        /// Radio player widget
```
Comment (lewandowski-jan) [inline mobile/lib/main.dart:260]: don't write comments when they do not add anything to the code written

File: mobile/lib/main.dart [hunk 5]
```diff
@@ -29,6 +30,7 @@ class MainApp extends HookWidget {
   MainApp({super.key});
 
   final _scaffoldKey = GlobalKey<ScaffoldState>(debugLabel: 'Inner scaffold');
+  // ignore: unused_field
```
Comment (lewandowski-jan) [inline mobile/lib/main.dart:33]: if it's unused you can probably remove it

File: mobile/lib/components/utility/shadowed_container.dart [hunk 6]
```diff
@@ -31,12 +35,13 @@ class ShadowedContainer extends StatelessWidget {
...
       padding: padding,
       decoration: BoxDecoration(
-        color: shadowColor,
-        boxShadow: const [
+        color: backgroundColor,
+        shape: shape ?? BoxShape.rectangle,
+        boxShadow: [
           BoxShadow(
-            color: Colors.black38,
+            color: shadowColor ?? Colors.black38,
```
Comment (lewandowski-jan) [inline mobile/lib/components/utility/shadowed_container.dart:42]: shouldn't we use palette colors https://github.com/RadioAktywne/ra-app/blob/master/mobile/lib/resources/colors.dart?

File: mobile/lib/components/utility/refreshable_fetch_widget.dart [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+/// Sets up hooks needed for fetching, loading and
+/// re-fetching data.
+///
+/// This function is a hook and should be used as such -
+/// at the top of the widget's `build()` function.
+///
+/// [defaultValue] - default value of the fetched data.
+///
+/// [fetchFunction] - function for fetching and re-fetching data
+_RefreshableFetchController<T> _useRefreshableFetchController<T>(
```
Comment (lewandowski-jan) [inline mobile/lib/components/utility/refreshable_fetch_widget.dart:92]: haven't used hooks for a while but I think you could do this easier

**Review by lewandowski-jan — DISMISSED** [review]
Looks good generally, but I'd advise to do here colors and texts refactor or create separate tasks for it

File: mobile/lib/resources/fetch_data.dart (hunk 1 above)
Comment (julianuziemblo) [reply mobile/lib/resources/fetch_data.dart:18]: Good idea for the future, I will create the tasks for them

File: mobile/lib/pages/plyta_tygodnia_page.dart (hunk 3 above)
Comment (julianuziemblo) [reply mobile/lib/pages/plyta_tygodnia_page.dart]: Good tip, just added that

File: mobile/lib/components/utility/refreshable_fetch_widget.dart (hunk 7 above)
Comment (julianuziemblo) [reply mobile/lib/components/utility/refreshable_fetch_widget.dart:92]: It'd be simple with 1 fetch, the reloads make it more complicated tho, that's why the `useMemoized` and `useFuture` are also used

File: mobile/lib/components/radio_player/radio_player_widget.dart [hunk 8]
```diff
@@ -8,6 +8,7 @@ import 'package:radioaktywne/state/audio_handler_cubit.dart';
 import 'package:text_scroll/text_scroll.dart';
 
 /// The radio player, consisting of a [RaPlayButton]
+/// and the stream title.
```
Comment (lewandowski-jan) [inline mobile/lib/components/radio_player/radio_player_widget.dart:11]: weird comment

File: mobile/lib/components/radio_player/radio_player_widget.dart [hunk 9]
```diff
@@ -143,6 +144,7 @@ class _StreamPlayButton extends StatelessWidget {
   }
 }
 
+/// The display of the radio stream title.
```
Comment (lewandowski-jan) [inline mobile/lib/components/radio_player/radio_player_widget.dart:147]: doubled?

**Review by lewandowski-jan — APPROVED** [review]

**Review by lewandowski-jan — APPROVED** [review]

**Pull request merged**